test/select.jl

```diff
@@ -0,0 +1,838 @@
+module TestSelect
+
+using DataFrames, Test, Random
+
+Random.seed!(1234)
+
+@testset "select! Not" begin
+    df = DataFrame(a=1, b=2, c=3, d=4, e=5)
+    @test_throws BoundsError select!(df, Not(0))
+    @test_throws BoundsError select!(df, Not(6))
+    @test_throws ArgumentError select!(df, Not([1, 1]))
+    @test_throws ArgumentError select!(df, Not(:f))
+    @test_throws BoundsError select!(df, Not([true, false]))
+
+    d = copy(df)
+    select!(d, Not([:a, :e, :c]))
+    @test names(d) == [:b, :d]
+    select!(d, Not(:b))
+    @test d == DataFrame(d=4)
+    DataFrames._check_consistency(d)
+
+    d = copy(df)
+    select!(d, Not(r"[aec]"))
+    @test names(d) == [:b, :d]
+    select!(d, Not(r"b"))
+    @test d == DataFrame(d=4)
+
+    d = copy(df)
+    select!(d, Not([2, 5, 3]))
+    @test names(d) == [:a, :d]
+    select!(d, Not(2))
+    @test d == DataFrame(a=1)
+
+    d = copy(df)
+    select!(d, Not(2:3))
+    @test d == DataFrame(a=1, d=4, e=5)
+
+    d = copy(df)
+    select!(d, Not([false, true, true, false, false]))
+    @test d == DataFrame(a=1, d=4, e=5)
+end
+
+@testset "select Not" begin
+    df = DataFrame(a=1, b=2, c=3, d=4, e=5)
+    @test_throws BoundsError select(df, Not(0))
+    @test_throws BoundsError select(df, Not(6))
+    @test_throws ArgumentError select(df, Not([1, 1]))
+    @test_throws ArgumentError select(df, Not(:f))
+    @test_throws BoundsError select(df, Not([true, false]))
+
+    df2 = copy(df)
+    d = select(df, Not([:a, :e, :c]))
+    @test names(d) == [:b, :d]
```

Comment:
This is redundant with the next line. Same below.

Reply:
fixed